wagtail/models/sites.py

```diff
@@ -1,48 +1,60 @@
 from collections import namedtuple
+from copy import deepcopy
+from typing import List, Tuple, Union
 
 from django.apps import apps
 from django.conf import settings
-from django.core.cache import cache
 from django.core.exceptions import ValidationError
 from django.db import models
-from django.db.models import Case, IntegerField, Q, When
 from django.db.models.functions import Lower
 from django.http.request import split_domain_port
 from django.utils.translation import gettext_lazy as _
 
+from wagtail.utils.local import ClearableLocal
+
 MATCH_HOSTNAME_PORT = 0
 MATCH_HOSTNAME_DEFAULT = 1
 MATCH_DEFAULT = 2
 MATCH_HOSTNAME = 3
 
 
+_sites_cache = ClearableLocal()
```

Comment:
Issue: We want cache hits to be as high as possible, so we actually want the cache to be shared as much as possible, rather than "local". A global variable, so long as mutations were atomic, is probably closer to what we want here.

Reply:
@RealOrangeOne Let's remind ourselves about the current state of things:

1. The data is currently cached for the current `HttpRequest`
2. In order to access the data efficiently, the `HttpRequest` needs to be passed around everywhere, because thats where the references to cached data are stored
3. There is absolutely no invalidation of the cached data (it lives on for the duration of the request)

All I'm really changing here is where the data is stored...  putting it in a place where it can be accessed regardless of whether you have access to the `HttpRequest`. Each 'request' is handled by a separate thread in Django, so caching via `asgiref.local.Local` is equivalent to caching for the request.

Reply:
Putting the cached data in a central/known place also gives us the added benefit of being able to invalidate cached data, which is another massive bonus

Reply:
I would highly recommend taking a look at the code for Local here. It's a fantastic bit of code: https://github.com/django/asgiref/blob/main/asgiref/local.py

It's the same mechanism Django itself uses for caching the active/current language for the current user/request, without you needing access to the `HttpRequest`.